### PR TITLE
HERD name change

### PR DESCRIPTION
**Pull request by mavaylon1**
## Motivation

What was the reasoning behind this change? Please explain the changes briefly.
Updates `ExternalResources` to have a uniform name throughout the codebase and the literature, which is now "HERD".
## How to test the behavior?
```
Show how to reproduce the new behavior (can be a bug fix or a new feature)
```
Make sure unit tests and docs pass.
## Checklist

- [ ] Did you update CHANGELOG.md with your changes?
- [ ] Have you checked our [Contributing](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst) document?
- [ ] Have you ensured the PR clearly describes the problem and the solution?
- [ ] Is your contribution compliant with our coding style? This can be checked running `ruff` from the source directory.
- [ ] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/hdmf-dev/hdmf/pulls) for the same change?
- [x] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

**Comment by mavaylon1** [comment]
Fix #932

**Comment by mavaylon1** [comment]
Note:
1. Merge TermSet first to avoid (mostly avoid) merge conflicts
2. Merge the schema first then this PR

File: docs/gallery/plot_external_resources.py [hunk 1]
```diff
@@ -1,34 +1,34 @@
 """
-ExternalResources
+HERD
 =================
```
Comment (oruebel) [inline docs/gallery/plot_external_resources.py]:
```suggestion
HERD: HDMF External Resources Data Structure
==============================================
```

**Comment by mavaylon1** [comment]
I did another pass and found some areas that didn't make the shift. I fixed them so I do believe we are good to go. Once this is merged, then I will do a fix on pynwb.

**Comment by mavaylon1** [comment]
The schema has been released as well.

**Review by oruebel — APPROVED** [review]
Looks good to me

**Pull request merged**